### DISCUSSION TITLE
Ash measurement

### DISCUSSION THREAD

**Discussion by theb1rn** [dominikamann/oekofen-pellematic-compact#39] [Ideas]
Hi all,
I just noticed by browsing through my settings that there is the possibility to enable an ash measurement which counts the amount of ash that goes into my ashbox. This is availablewhen pellet consumption is calculated with the "ZWB method" (sorry - I only know the german description for it) and ash-counting is set to yes in the "ash" menu item under PES.

I was wondering if you found the same and if it is available in the JSON.
In my JSON it is not available. I'm running a PES20 with touch 4.02a_P

If it is not part of the JSON - is there a way to get the data into home assistant? Would be supper nice to have it there to trigger a reminder to empty the ashbox before it produces an error message.

![oekofen_ash](https://github.com/dominikamann/oekofen-pellematic-compact/assets/79479330/4d14e22e-6194-4d11-bc14-7146beae94fb)

**Comment by dominikamann** [comment]
Hello @theb1rn,

I also use this functionality with my Pellematic. Unfortunately the info is not available via the JSON interface. I think there is no way to make it work with HomeAssistant until Ökofen updates the interface.

Workaround: Calculate/Measure the pellet consumption until the ashbox needs to be cleaned. E.g. 250kg pellets was burned then the box is full. Create a numerical helper entity and set it to e.g. 220kg. Then you can daily automatically subtract your daily pellet consumption. With an automation check if the helper entity gets below zero (value is <= 0) if so, send a notification 🔔 "Ash box will soon be full". After cleaning, set your custom helper entity back to 220kg. Can also be done via an "Ash box cleaned" button in Dashboard, for example.

Dominik

**Reply by theb1rn** [reply]
Thanks. Thats a great idea. Still fiddling with the pellet consumption measurements within my Oekofen - once the data is more accurate I'll share my pellet to ash ratio

**Reply by Azygos64** [reply]
Zuerst einmal vielen Dank für die Integration - wirklich genial!!!
Vielleicht habe ich einen Draht zu Ökofen...
Doch vorher wollte ich gerne wissen, ob es das Feld inzwischen vielleicht schon gibt. In der App wird es jedenfalls angezeigt - also muss der Wert irgendwie aus dem Ofen ins Internet oder anders in die App kommen. Meine aktuelle Firmware ist 4.02b.
Ansonsten könnte ich mal bei Ökofen "nachhaken"...

**Reply by dominikamann** [reply]
Immer gerne meines Wissens nach nicht. Viele Grüße
